### ISSUE TITLE
Exception displayed when launching the the jekyll serve command

### ISSUE THREAD

**Issue by FrontRunnerUp** [mmistakes/skinny-bones-jekyll#24]
Hello,
Foremost thank you for the nice theme you provided for usage with such a convenient license!

I am new to jekyll, having come from a fairly recent HTML/CSS/JS background, as well as ASP NET MVC and Angular. I liked this theme for my learning and own usage (if succeed in going through the learning curve)

I've just come across a problem. Below the explanation of my process:

I recently installed ruby, rubygems, and jekyll in a Windows 8.1 (x64) box, ran jekyll, created a new website from scratch, and everything went just fine.
Then I found your theme, I cloned it locally to my box, and started tinkering with it.
My first problem was that I installed bundler and the theme's dependencies globally, and most of the gems I had from previous jekyll installation got duplicated to a lesser version. Could not get the jekyll serve command to work and first thought about the duality in versions for some gems.

I reasoned that uninstalling all dependencies installed with bundler, repairing the latest jekyll I had - which is version 2.5.3 at present - and installing dependencies by project would fix the issue. I followed this [stackoverflow answer on bundler](http://stackoverflow.com/a/21384876/1960071) to install bundler's dependencies by project instead (not globally).
So for this jekyll theme I installed all gem dependencies using bundler, on the vendor\bundle folder.
It added all of them, including jekyll 2.4.0.

The problem was not fixed, there are two parts to it:

First, when executing `jekyll serve`, there is an exception displayed, related to mercenary gem (below the stack trace). You will notice is trying to use the global jekyll 2.5.3 gem:

E:\jekyll-Sites\portfolio [master +9 ~5 -0 !]> `jekyll serve --watch`
Configuration file: E:/jekyll-Sites/portfolio/_config.yml
C:/Ruby21-x64/lib/ruby/gems/2.1.0/gems/jekyll-2.5.3/lib/jekyll/plugin_manager.rb:29:in `require': cannot load such file
-- jekyll-sitemap (LoadError)
        from C:/Ruby21-x64/lib/ruby/gems/2.1.0/gems/jekyll-2.5.3/lib/jekyll/plugin_manager.rb:29:in `block in require_gems'
        from C:/Ruby21-x64/lib/ruby/gems/2.1.0/gems/jekyll-2.5.3/lib/jekyll/plugin_manager.rb:26:in `each'
        from C:/Ruby21-x64/lib/ruby/gems/2.1.0/gems/jekyll-2.5.3/lib/jekyll/plugin_manager.rb:26:in `require_gems'
        from C:/Ruby21-x64/lib/ruby/gems/2.1.0/gems/jekyll-2.5.3/lib/jekyll/plugin_manager.rb:19:in `conscientious_require'
        from C:/Ruby21-x64/lib/ruby/gems/2.1.0/gems/jekyll-2.5.3/lib/jekyll/site.rb:79:in `setup'
        from C:/Ruby21-x64/lib/ruby/gems/2.1.0/gems/jekyll-2.5.3/lib/jekyll/site.rb:41:in `initialize'
        from C:/Ruby21-x64/lib/ruby/gems/2.1.0/gems/jekyll-2.5.3/lib/jekyll/commands/build.rb:29:in `new'
        from C:/Ruby21-x64/lib/ruby/gems/2.1.0/gems/jekyll-2.5.3/lib/jekyll/commands/build.rb:29:in `process'
        from C:/Ruby21-x64/lib/ruby/gems/2.1.0/gems/jekyll-2.5.3/lib/jekyll/commands/serve.rb:26:in `block (2 levels) in init_with_program'
        from C:/Ruby21-x64/lib/ruby/gems/2.1.0/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `call'
        from C:/Ruby21-x64/lib/ruby/gems/2.1.0/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `block in execute'
        from C:/Ruby21-x64/lib/ruby/gems/2.1.0/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `each'
        from C:/Ruby21-x64/lib/ruby/gems/2.1.0/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `execute'
        from C:/Ruby21-x64/lib/ruby/gems/2.1.0/gems/mercenary-0.3.5/lib/mercenary/program.rb:42:in `go'
        from C:/Ruby21-x64/lib/ruby/gems/2.1.0/gems/mercenary-0.3.5/lib/mercenary.rb:19:in `program'
        from C:/Ruby21-x64/lib/ruby/gems/2.1.0/gems/jekyll-2.5.3/bin/jekyll:20:in `<top (required)>'
        from C:/Ruby21-x64/bin/jekyll:23:in `load'
        from C:/Ruby21-x64/bin/jekyll:23:in `<main>'

Second, when running the `bundle exec jekyll serve` command, for some strange reason the command is trying to call the post markdown file template located in the vendor\bundle jekyll 2.4.0 path:

E:\jekyll-Sites\portfolio [master +9 ~5 -0 !]> `bundle exec jekyll serve`
DL is deprecated, please use Fiddle
Configuration file: E:/jekyll-Sites/portfolio/_config.yml
            Source: E:/jekyll-Sites/portfolio
       Destination: E:/jekyll-Sites/portfolio/_site
      Generating...
             ERROR: YOUR SITE COULD NOT BE BUILT:
                    ------------------------------------
                    Invalid date '0000-00-00': Post '/vendor/bundle/ruby/2.1.0/gems/jekyll-2.4.0/lib/site_template/_post
s/0000-00-00-welcome-to-jekyll.markdown.erb' does not have a valid date in the filename.

Here is the image of the above:
![image](https://cloud.githubusercontent.com/assets/3148402/6435488/5ca5feb0-c075-11e4-8c97-5ccc3a748779.png)

I already have posts files hosted on the theme, but they do not get referenced for serving the site.
I think this may not be a theme issue, but surely wanted to let know in order to get more light into this issue, and even get a resolution for this.

Thanks for your help in advance!

**Comment by mmistakes** [comment]
That's really weird, though not surprising since Windows is notorious for causing a ruckus. I don't think you have tinker at all with your posts or the theme, it's totally a gem conflict/issue on Windows. Which gem that is can be a problem to track down because the error handling isn't always specific.

The theme doesn't require any additional gems except the ones Jekyll specifies. What's probably going on is one of the versions required in `Gemfile.lock` is probably different or didn't install properly on your machine, so Jekyll barfs when you try to execute it.

Did you do a `bundle update` and observe any errors? For giggles you could try and remove both `Gemfile` and `Gemfile.lock` and see if Jekyll runs.

The only other thing I could think of that I see occasional problems with is the Octopress gem. You can strip that out of everything since it's completely optional. I only included it because people kept writing Rake and Thor tasks as generators for creating new posts and pages. Octopress seemed way more robust than that so here we are.

**Comment by mmistakes** [comment]
Also it should be fairly easy to pare the theme down to "vanilla" Jekyll. In the `Gemfile` I included dependencies for [Bourbon](http://bourbon.io/) and [Neat](http://neat.bourbon.io/). Neither of these gems are needed since I'm already bundling the Sass partials them come with directly in the theme. The only reason you'd need the gems installed is if you wanted to update the `.scss` files that come with them manually.

[Octopress](https://github.com/octopress/octopress) I already mentioned so that could be removed safely. And then there is [Jekyll sitemap](https://github.com/jekyll/jekyll-sitemap). I noticed an error for that in the first part of your issue. That plugin is written by the Jekyll folks so there shouldn't be any extraneous dependencies, but you never know. As a test you could remove that as well, it only builds a `sitemap.xml` which could be done in other ways if the gem causes you problems.

So to summarize. I'd remove `Gemfile` and `Gemfile.lock`, make sure Jekyll is updated to the latest version, and run it the prescribed way: `jekyll build` and `jekyll serve`. 

I wouldn't worry too much about older gems being installed. That's the point of using Bundler and having a Gemfile. It manages what versions you should be using with your project. If you remove the Gemfile and execute the normal way the latest versions will be used by default.

I've definitely seen the errors you're having before so it's not something uncommon. Getting gems to play nice on Windows can be a pain sometimes. If you still have problems I'd bump around the closed issued on the Jekyll repo or even my other theme's repos. I could have sworn I've provided these solutions before :wink:

**Comment by FrontRunnerUp** [comment]
Thank you for the prompt responses!
I was into reviewing the guidelines provided in this thread and also in other issues already posted.  Indeed, some of the problems I came across are addressed in other issues that are closed.

So far I can tell that I did not have any problems with the octopress gem. In fact it worked swiftly when generating new posts. 
Removing the gemfile and gemfile.lock archives do not prevent the second issue from happening (this time is shown when using `jekyll serve` command.instead of bundle)
I am yet to review the sitemap issue, 
Since I am fairly new on this, I am into reviewing the Gemfile.lock details so as to learn and maybe pinpoint the issue. My aim is to not take too much out from your setup, after all, it worked in your machine. :)
I'll give any update on what I find. Thanks again!

**Comment by mmistakes** [comment]
Sounds good. I've gotten it to work on a Windows 7 setup, so it's possible. I previously had to do a lot of sorting out of gems and such and really can't remember what I changed. But I do know Pygments.rb (that does syntax highlighting) is a real big pain point with Windows and often gives cryptic errors that appear to be unrelated to it. 

Do you have posts with code in them or are you using the demo posts? Sounds to me Markdown is having problems with some of the content. I only ask because that "welcome to Jekyll" post in the error you posted above has code in it. There's a good chance Markdown or the syntax highlighter is having problems.

Some other things to try:

1. Remove all the posts and see if the site builds.
2. Triple check your `_config.yml` to make sure you don't have invalid YAML. Spaces/tabs matter and can blow things up.

**Comment by FrontRunnerUp** [comment]
Some updates:
 - I changed `highlighter: rouge` (I saw [in this tutorial](http://jekyll-windows.juthilo.com/3-syntax-highlighting/) that I needed to install python for the pygments to work, so just changed it from the get go, although I noticed that **pygments.rb** in the project doesn't seem a python file)
- I checked my .yml and .md posts with an [online yml linter](http://www.yamllint.com/), and all green.
- I attempted to serve the site without any posts files (.md) and would not serve.

I am targeting the issue around the specific gem dependencies I installed on my portfolio project.
I will be testing -among others- editing the gemfile to not require the potentially problematic gems (using [some bundler parameters](http://bundler.io/gemfile.html) i.e. `gem 'rspec', :require => false` )
Still sifting through documentation.

**Comment by FrontRunnerUp** [comment]
More updates.
- Changed back the highlighter to pygments.rb (there was an error with rouge, my huntch is that it was not installed for the specific project)

**Finally got the server to serve the web page successfully!**

I noticed a pattern in the errors that were displaying: they referred almost exclusively to test files included with the gems that were of type .md, html, even scss.
After I removed the containing folders of the files, site was served successfully.

The removed gem test directories inside my vendor\bundle dir include: 
- jekyll-2.4.0\lib\site_template
- octopress-3.0.0.rc.15\test
- octopress-3.0.0.rc.34\local (updated gem through `bundle update` command)
- octopress-3.0.0.rc.34\site (updated gem through `bundle update` command)
- octopress-docs-0.0.9\docs
- octopress-escape-code-1.0.2\test
- octopress-escape-code-2.0.6\test (updated gem through `bundle update` command)

I'm yet to measure the impact (if any) of these removals on the project. If I encounter any issues will let you know.
I am yet to pinpoint how to strip down the project-specific gem installation, but that's a meal for another time.

I'm closing this issue. 
Thanks!

**Comment by mmistakes** [comment]
That's a strange one. I'd follow up with the gem creators to let them know
of the issues. Just in case their bugs on their end.

**Comment by FrontRunnerUp** [comment]
It is weird indeed.
Will be logging the issue within the jekyll repo.
